### ISSUE TITLE
Default Enum value is always used instead of the one passed in request

### ISSUE THREAD

**Issue by dbronecki** [theiterators/kebs#2]
I'm using enum as parameter with default value. Default value is always used instead of the one I passed to request.

Test case:
```scala
import org.scalatest.{Matchers, WordSpec}
import akka.http.scaladsl.testkit.ScalatestRouteTest
import akka.http.scaladsl.server._
import Directives._
import enumeratum.{Enum, EnumEntry}

sealed trait SortOrder extends EnumEntry
object SortOrder extends Enum[SortOrder] {
  case object Asc extends SortOrder
  case object Desc extends SortOrder

  override val values = findValues
}

class KebsTest extends WordSpec with Matchers with ScalatestRouteTest {

  import pl.iterators.kebs.unmarshallers._
  import enums._

  val smallRoute =
    get {
      path("test_enum") {
        parameter('sort.as[SortOrder] ? SortOrder.Desc) { sort =>
          complete {
            s"Sort was $sort"
          }
        }
      }
    }

  "Kebs" should {

    "work with default enum values" in {
      Get("/test_enum?sort=Asc") ~> smallRoute ~> check {
        responseAs[String] shouldBe "Sort was Asc"
      }
    }
  }
}
```

**Comment by marcin-rzeznicki** [comment]
Thank you for the analysis. So, as you discovered, it is because the `?` method changes type of `NameReceptacle` (it is defined as `def ?[B](default: B) = new NameDefaultReceptacle(name, default)`) - so, ong story short - the `Unmarshaller` that is materialized is `FromStringUnmarshaller[SortOrder.Desc.type]` - provided by kebs for `case object`s. The unmarshaller for case object ignores its argument and returns constant object - which is the direct cause for the bug. Thanks @dbronecki for analysis! I fixed that in 71ad0b2 by removing case objects' unmarshallers - it doesn't look like a feature that is needed. I think that it was worth sacrificing this unneeded feature for less surprising behavior. Let me know wdyt @dbronecki !

**Comment by dbronecki** [comment]
Yeah, it's better now - no surprising behaviour. I have no idea how to fix it otherwise, so for now we can mark this issue as resolved.
